rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -0,0 +1,345 @@
+        }
+
+        # If no exceptions are thrown, we have a valid connection
+        self._post_payload(graph_ql_payload)
+        return True
+
+    #################
+    # Private Methods
+    #################
+
+    def _get_org_key(self) -> str:
+        """
+        Get the org key from GraphQL
+
+        :return: String
+        """
+
+        payload = {
+            "query": "{ organizations(first: 1) { edges { node { id name } } } }"
+        }
+        result_object = self._post_payload(payload)
+        self.logger.info("Organization ID query complete.")
+        return result_object.get("data").get("organizations").get("edges")[0].get("node").get("id")
+
+    def _get_headers(self) -> dict:
+        """
+        This build and returns headers for the request session
+
+        :return: dict
+        """
+        return {
+            "X-Api-key": self.api_key,
+            "Accept-Version": "kratos",
+            "Accept": "*/*",
+            "Accept-Encoding": "gzip, deflate, br"
+        }
+
+    def _post_payload(self,
+                      payload: dict
+                      ) -> dict:
+        """
+        This will post a given payload to the API using the connection session
+
+        :param payload: dict
+        :return: dict
+        """
+        result = self.session.post(self.endpoint, json=payload)
+
+        try:
+            result.raise_for_status()
+        except Exception:
+            raise APIException(cause="Error connecting to the Insight Agent API.",
+                               assistance="Please check your Organization ID and API key.\n",
+                               data=result.text)
+
+        results_object = result.json()
+
+        if results_object.get("errors"):
+            raise APIException(cause="The Insight Agent API returned errors",
+                               assistance=results_object.get("errors"))
+
+        return results_object
+
+    def _setup_endpoint(self,
+                        region_string: str
+                        ) -> str:
+        """
+        Creates the URL endpoint for the API based on the region
+
+        :param region_string: string
+        :return: string
+        """
+        region_code = region_map.get(region_string)
+
+        if region_code:
+            endpoint = f"https://{region_code}.api.insight.rapid7.com/graphql"
+        else:
+            # It's an enum, hopefully this never happens.
+            raise APIException(cause="Region not found.",
+                               assistance="Region code was not found for selected region. Available regions are: United States, "
+                                          "Europe, Canada, Australia, Japan")
+
+        return endpoint
+
+    def _get_agent(self,
+                   agent_input: str,
+                   agent_type: str
+                   ) -> dict:
+        """
+        Gets an agent from a Mac, IP, or Hostname.
+
```

Comment:
:param agent_input: MAC address, IP address, or hostname for the agent
:param agent_type: ?? what is agent type?